curp/src/server/curp_node.rs

```diff
@@ -731,49 +527,51 @@ impl<C: 'static + Command, RC: RoleChange + 'static> CurpNode<C, RC> {
+                                    hb_opt = true;
+                                } else {
+                                    debug!("ae rejected by {}", connect.id());
+                                }
+                                // Check Follower shutdown
+                                // When the leader is in the shutdown state, its last log must be shutdown, and if the follower is
+                                // already synced with leader and current AE is a heartbeat, then the follower will commit the shutdown
+                                // log after AE, or when the follower is not synced with the leader, the current AE will send and directly commit
+                                // shutdown log.
+                                if is_shutdown_state
+                                    && ((curp.is_synced(connect_id) && is_empty)
+                                        || (!curp.is_synced(connect_id) && is_commit_shutdown))
+                                {
+                                    if let Err(e) = connect.trigger_shutdown().await {
+                                        warn!("trigger shutdown to {} failed, {e}", connect_id);
+                                    } else {
+                                        debug!("trigger shutdown to {} success", connect_id);
+                                    }
                                     break false;
                                 }
                             }
-                        } else {
-                            hb_opt = true;
-                        }
-                        // Check Follower shutdown
-                        // When the leader is in the shutdown state, its last log must be shutdown, and if the follower is
-                        // already synced with leader and current AE is a heartbeat, then the follower will commit the shutdown
-                        // log after AE, or when the follower is not synced with the leader, the current AE will send and directly commit
-                        // shutdown log.
-                        if is_shutdown_state
-                            && ((curp.is_synced(connect_id) && is_empty)
-                                || (!curp.is_synced(connect_id) && is_commit_shutdown))
-                        {
-                            if let Err(e) = connect.trigger_shutdown().await {
-                                warn!("trigger shutdown to {} failed, {e}", connect_id);
-                            } else {
-                                debug!("trigger shutdown to {} success", connect_id);
+                            Err(err) => {
+                                warn!("ae to {} failed, {err:?}", connect.id());
+                                if is_shutdown_state {
+                                    ae_fail_count += 1;
+                                    if ae_fail_count >= 5 {
+                                        // ae_fail_count = 0; RESET?
```

Comment:
We don't need to reset `ae_fail_count`, Because each `sync_follower_task` can only shutdown once.

Reply:
Ok, I'll remove this comment in the next PR due to interruption the approve.